### DISCUSSION TITLE
[1 febbraio 2024] Esercizio 3

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/psd2223#265] [Esami]
Codificare $\Large X = -129,125$ nello standard IEEE 754 half-precision.

Calcolare poi la somma $\Large Z=X+Y$, dove $\Large Y =<0;00110;1110001100>$ e dare la rappresentazione di $\Large Z$ nel formato IEEE 754 half-precision.

Infine convertire i bit di $\Large Z$ in base $\Large 16$.

**Comment by Nw4pS** [comment]
@FeddyLix17 potresti postare la risoluzione dell'esercizio? non capisco come fare la somma tra quei due numeri, non posso trasformare il numero in negativo in ca2 perché non avrebbe più lo stesso esponente in quanto dovrei cambiare la virgola, potrei provare a sottrarre direttamente i numeri, ma niente. non riesco neanche a trasformare y in decimale poiché mi ritrovo a calcolare 2^(-8) + 2^(-9) + 2^(-10) + 2^(-11) + 2^(-15) + 2^(-16) e senza calcolatrice (all'esame non può essere portata) non è fattibile.

**Comment by FeddyLix17** [comment]
$$
    \Large
    \begin{multline}
         X = -129, 125_{10}\\
         \\
         129, 125_{10} = 10000001,001_2 = 1,0000001001_2 * 2^7
    \end{multline}
$$

<div align="center">
<table>
  <tr>
    <th colspan="2" rowspan="2">X</th>
    <th>Segno</th>
    <th colspan="5">Esponente (bias +15)</th>
    <th colspan="10">Mantissa</th>
  </tr>
  <tr>
    <td align="center">1</td>
    <td>1</td>
    <td>0</td>
    <td>1</td>
    <td>1</td>
    <td>0</td>
    <td>0</td>
    <td>0</td>
    <td>0</td>
    <td>0</td>
    <td>0</td>
    <td>0</td>
    <td>1</td>
    <td>0</td>
    <td>0</td>
    <td>1</td>
  </tr>
</table>

<table>
  <tr>
    <th colspan="2" rowspan="2">Y</th>
    <th>Segno</th>
    <th colspan="5">Esponente (bias +15)</th>
    <th colspan="10">Mantissa</th>
  </tr>
  <tr>
    <td align="center">0</td>
    <td>0</td>
    <td>0</td>
    <td>1</td>
    <td>1</td>
    <td>0</td>
    <td>1</td>
    <td>1</td>
    <td>1</td>
    <td>0</td>
    <td>0</td>
    <td>0</td>
    <td>1</td>
    <td>1</td>
    <td>0</td>
    <td>0</td>
  </tr>
</table>
</div>

<br>

$$
    \Large Y = 1,1110001100 * 2^{6-15=-9} = 0, \overbrace{\underbrace{00000000}_{\texttt{mantissa} \atop \texttt{half-precision}}0000000}^{15}11110001100 * 2^7
$$

> [!NOTE]
> la bassa accuratezza dell'half-precision tronca tutta la parte rilevante della mantissa di Y una volta allienati gli esponenti, per tanto
>
> $$\Large Z = X_{\texttt{half-precision}} + Y_{\texttt{half-precision}} \approx X $$
>
> di fatto $\Large Y_{10} = 0.003685$ e $\Large Z = -129, 125 + 0.003685 \approx - 129, 122$ (dove in half-precision tutta la parte decimale verrebbe trascurata).

$$
    \Large Z = {\underbrace{1\ 101}_D\underbrace{10\ 00}_8\underbrace{0000}_0\underbrace{1000}_8} = 0xD808\_{16}
$$
